grails-app/domain/org/pih/warehouse/picklist/PicklistItem.groovy

```diff
@@ -15,11 +15,16 @@ import org.pih.warehouse.inventory.InventoryItem
 import org.pih.warehouse.inventory.RefreshProductAvailabilityEvent
 import org.pih.warehouse.order.OrderItem
 import org.pih.warehouse.requisition.RequisitionItem
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.ApplicationEventPublisher
 
 class PicklistItem implements Serializable {
 
+    @Autowired
+    ApplicationEventPublisher applicationEventPublisher
+
     def publishRefreshEvent = {
-        Holders.grailsApplication.mainContext.publishEvent(new RefreshProductAvailabilityEvent(this))
+        applicationEventPublisher?.publishEvent(new RefreshProductAvailabilityEvent(this))
```

Comment:
I think this refactor was not needed. Was it?

Reply:
You are right. Previously, without those changes, I got an error about it. For now, I reverted those changes and it works fine, thanks.